nullaway/src/test/java/com/uber/nullaway/TypeUseAnnotationsTests.java

```diff
@@ -189,21 +193,148 @@ public void typeUseAnnotationOnInnerMultiLevel() {
+            "public @interface Nullable {}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "class A { class B { class C {} } }",
+            "class Test {",
+            "  // ok, treated as declaration",
+            "  @Nullable A.B.C foo1 = null;",
+            "  // not ok, invalid location for both type-use and declaration",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  A.@Nullable B.C foo2 = null;",
+            "  // ok, treated as type-use",
+            "  A.B.@Nullable C foo3 = null;",
+            "  // ok, treated as declaration",
+            "    @Nullable A.B foo4 = null;",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseAndDeclarationAnnotationOnInnerMultiLevelLegacy() {
+    makeLegacyModeHelper()
+        .addSourceLines(
+            "Nullable.java",
+            "package com.uber;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Target;",
+            "@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.TYPE_USE})",
+            "public @interface Nullable {}")
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "class A { class B { class C {} } }",
+            "class Test {",
+            "  // ok, treated as declaration",
+            "  @Nullable A.B.C foo1 = null;",
+            "  // ok, treated as type-use",
+            "  A.@Nullable B.C foo2 = null;",
+            "  // ok, treated as type-use",
+            "  A.B.@Nullable C foo3 = null;",
+            "  // ok, treated as declaration",
+            "    @Nullable A.B foo4 = null;",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseAnnotationOnMethodReturnType() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.checkerframework.checker.nullness.qual.Nullable;",
+            "class A { class B { class C {} } }",
+            "class Test {",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  public @Nullable A.B.C method1() { return null; }",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  public A.@Nullable B.C method2() { return null; }",
+            "  public A.B.@Nullable C method3() { return null; }",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  public @Nullable A.B method4() { return null; }",
+            "  public @Nullable A method5() { return null; }",
+            "  public @Nullable int method6() { return 0; }",
```

Comment:
remove this one?  eventually this should be an error (but not for this PR).  `@Nullable` makes no sense on a primitive type